File: android/lab7/FM/app/src/main/java/com/mylab/topfm/ui/MainActivity.java
package com.mylab.topfm.ui;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.LifecycleOwner;
import androidx.lifecycle.ViewModelProvider;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.Spinner;
import android.widget.Toast;

import com.mylab.topfm.Track;
import com.mylab.topfm.TrackAdapter;
import com.overwees.topfm.R;
import com.mylab.topfm.TrackViewModel;
import com.mylab.topfm.service.DBHelper;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    DBHelper dbHelper;
    Context context;
    Spinner spinner;
    ListView listView;
    String artist = "Rammstein";
    ArrayList<Track> trackList;
    TrackAdapter adapter;
    TrackViewModel model;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        dbHelper = new DBHelper(this);
        context = this;
        spinner = (Spinner)findViewById(R.id.spinner);
        listView = (ListView) findViewById(R.id.listView);
        if(!isConnectingToInternet(getApplicationContext()))   {
            Toast.makeText(getApplicationContext(), "No internet access. Looking at internal saved data", Toast.LENGTH_LONG).show();
            trackList = dbHelper.getTopTenTracksByArtist(artist);
            adapter = new TrackAdapter(context,R.layout.track_activity, trackList);
            listView.setAdapter(adapter);
        } else{
            model = new ViewModelProvider(this).get(TrackViewModel.class);
            model.getValue(artist, context).observe((LifecycleOwner) context, value -> {
                adapter = new TrackAdapter(context,R.layout.track_activity, value);
                listView.setAdapter(adapter);
            });
            model.execute();
        }

        ArrayAdapter<CharSequence> artistsAdapter = ArrayAdapter.createFromResource(
                this,
                R.array.artists,
                android.R.layout.simple_spinner_item
        );
        AdapterView.OnItemSelectedListener itemSelectedListenerArtists = new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                if(isConnectingToInternet(getApplicationContext()))
                    model.getValue(artist, context).observe((LifecycleOwner) context, value -> {
                        adapter = new TrackAdapter(context,R.layout.track_activity, value);
                        listView.setAdapter(adapter);
                    });
                artist = (String)parent.getItemAtPosition(position);
                trackList = dbHelper.getTopTenTracksByArtist(artist);
                adapter = new TrackAdapter(context,R.layout.track_activity, trackList);
                listView.setAdapter(adapter);
            }

            @Override
            public void onNothingSelected(AdapterView<?> parent) {
            }
        };
        spinner.setAdapter(artistsAdapter);
        spinner.setOnItemSelectedListener(itemSelectedListenerArtists);


    }

    private boolean isConnectingToInternet(Context applicationContext){
        ConnectivityManager cm = (ConnectivityManager) getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo ni = cm.getActiveNetworkInfo();
        if (ni == null) {
            return false;
        } else
            return true;

    }
}

File: android/lab3/updateCalcVer/app/src/main/java/com/example/task3_3/MainActivity.kt
package com.example.task3_3

import Calc
import android.content.Intent
import android.os.Bundle
import android.widget.AdapterView.OnItemClickListener
import android.widget.ArrayAdapter
import android.widget.ListView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import java.util.*
import com.example.task3_3.R



class MainActivity : AppCompatActivity() {

    var options = arrayOf("Calculator", "History")

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main_horizontal)
        val adapter = ArrayAdapter(
            this,
            android.R.layout.simple_list_item_1,
            options
        )
        val listView = findViewById<ListView>(R.id.listView)
        listView.adapter = adapter
        listView.onItemClickListener =
            OnItemClickListener { parent, view, position, id ->
                when (position) {
                    0 -> startActivity(Intent(this@MainActivity, Calc::class.java))
                    1 -> startActivity(Intent(this@MainActivity, History::class.java))

                }
                Toast.makeText(
                    applicationContext, "Вы выбрали " +
                            parent.getItemAtPosition(position).toString(), Toast.LENGTH_SHORT
                ).show()
            }
    }
}

